docs/adr/0004-framework.md

```diff
@@ -0,0 +1,130 @@
+# Use React Framework
+
+* Status: proposed
+* Deciders: Monitor team
+* Date: 2022-05-03
+* Technical Story: [MNTOR-1588](https://mozilla-hub.atlassian.net/browse/MNTOR-1588)
+
+## Context and Problem Statement
+
+The Monitor team is tasked with launching a premium product offering as soon as possible. The current architecture of Monitor is not suited for synchronous feature development and lacks the complex client-side state management premium will require. 
+
+
+## Decision Drivers <!-- optional -->
+
+* The expansion in Monitor premium requires a framework to manage increased complexity and client-side state management
+* Promote component sharing between S&P web-based products, Relay & Monitor
+* Allow developers to work on specific components and features without affecting the rest of the codebase (fewer merge conflicts, makes it easier break down tickets)
+* Unidirectional (one-way) data flow makes it easier to understand the flow of data in the application, which may be necessary when handling multiple states (especially for the data removal sequence)
+* Monitor Front-end engineers have extensive React experience
+
+## Considered Options
+
+1. Migrate framework to React and Next.js
+1. Leave as-is 
+1. Use a different framework
+
+## Decision Outcome
+
+TBD
+
+*Note: If "Migrate to React/Next.js" is selected, we also need to decide on the migration plan*
+
+## Pros and Cons of the Options <!-- optional -->
+
+### Migrate framework to React and Next.js
+
+* Good, Component-based Architecture: This simplifies code maintenance by breaking complex UIs into reusable components.
+* Good, Testing: the virtual DOM makes it feasible to write non-brittle (i.e. without needing to fire up a full browser) unit tests of the front-end
+* Good, Fewer bugs: having React take care of removing, adding or changing DOM nodes removes a large surface area for bugs.
+* Good, Shared knowledge: This approach shares the same structure as Firefox Relay. This would immediately allow the team to reuse and reference.
+	* Note that FxA came to a similiar decision: [FxA ADR: Refactor Subscription Platform frontend - with Next.js](https://github.com/mozilla/fxa/blob/main/docs/adr/0035-refactor-payments-frontend-with-nextjs.md) 
+* Good, Enables static analysis: programmatically-generated DOM and explicit module imports enables optimal use of TypeScript to catch errors at compile-time and make code more self-explanatory.
+* Good, Industry Usage/Standard: React has a large and active community of developers, with many resources available, including libraries, tutorials, and support.
```

Comment:
Related advantage I just realised: this makes it easier for drive-by contributors to contribute, since they can use existing familiarity with React that they wouldn't have with our homegrown snowflake framework. (We've had a number of contributions to Relay's UI since it's been in React, which is awesome.)